### ISSUE TITLE
Model is not loading when a variant is defined on a mesh primitive other than the first one in the primitives array.

### ISSUE THREAD

**Issue by alexdaube** [google/model-viewer#3179]
### Description
Basically I have a model with a mesh that has 3 primitives defined in the array. Each of these primitives have a different material assigned. What I want to do is to apply a different variant mapping to each primitive. Since, I switched to model-viewer 1.10.x,  I can only put it on the first primitive in the array, otherwise it will not load the model(blank page with no errors in the console). I have tried to put variant mappings in all 3 primitives, or only on any of the other 2 primitives which yielded the same result.
I have not used model-viewer 1.9.x in production, so I am not certain if this is related to latest version or the one previous to that, but I can confirm that my model worked in 1.8. 
Any idea on what could cause this changed behavior? I would be willing to help if you guys think this is a model-viewer specific issue.

**Here are some screenshots of the GLTF file to kinda explain what I mean**:
Renders: 
![Screen Shot 2022-02-03 at 10 22 18 AM](https://user-images.githubusercontent.com/8660478/152373097-c6c2bb06-a9f0-4cda-ace1-23565ba38bb0.png)

Does not render:
![Screen Shot 2022-02-03 at 10 24 56 AM](https://user-images.githubusercontent.com/8660478/152373163-d13ef626-96bb-4c4c-85e9-4da3105ca70d.png)

![Screen Shot 2022-02-03 at 10 23 56 AM](https://user-images.githubusercontent.com/8660478/152373133-8cd6d38e-f1cd-41ee-af2d-2cce934c0386.png)

### Version
- model-viewer: v.1.8.0 - Works
- model-viewer: v.1.9.x - Not Tested 
- model-viewer: v1.10.1 -> Fails

### Browser Affected
- [X ] Chrome, version:  97.0.4692.99
- [ ] Edge
- [ ] Firefox
- [ ] IE
- [ ] Safari

### OS
- [ ] Android
- [ ] iOS
- [ ] Linux
- [ X] MacOS
- [ ] Windows

**Comment by elalish** [comment]
Interesting; are you setting variant materials using our editor, or a different tool (are you just hand-writing them in VSCode)? Can you link your model here so we can repro? Seems like it's likely a bug either in MV or in the glTF itself, though the screenshots look reasonable. Does this glTF render properly in other renderers (three.js editor, babylon sandbox, etc)?

**Comment by alexdaube** [comment]
@elalish, thanks for taking the time.
 
I have not used model-viewer's material creation tool, since I don't do this process in the browser just yet. I have an in-house python CLI that I usually use for this and I also tried out this lib(https://gltf-transform.donmccurdy.com/) to kinda see the differences that it would yield. I compared both resulted files, and they were both exactly the same.
Also, I have files with similar variant configurations which stopped working once I upgraded to 1.10.1 from 1.8.

As you suggested, I have tested my model in both the three.js editor and the babylon sandbox this morning. Both render properly and I can confirm that my variants work as intended in the babylon sandbox...

I don't think I can share my current files unfortunately. I will try to maybe cook something with similar variants that we can use to test this. In the meanwhile, I will try to start debugging and perhaps give a shot at fixing this on my side for sure.
